File: nextts/src/types/errorLog.ts
// src/types/errorLog.ts
export interface ErrorLog {
  errorLogId: number;
  errorTime: string;
  userName: string;
  errorNumber: number;
  errorSeverity?: number;
  errorState?: number;
  errorProcedure?: string;
  errorLine?: number;
  errorMessage: string;
    
  // Add the new properties to match C# model
  note?: string;                // Maps to Note in C# model
  lastUpdated?: string;         // DateTime stored as ISO string
  assignedTo?: string;          // Foreign key to AspNetUsers table
}

export interface ErrorLogQuery {
  text?: string;
  errorTimeFrom?: string;
  errorTimeTo?: string;
  errorSeverities?: number[];
  errorStates?: number[];
  pageSize?: number;
  pageIndex?: number;
  orderBy?: string;
    
  // Add new query parameters
  lastUpdatedFrom?: string;     // DateTime stored as ISO string
  lastUpdatedTo?: string;       // DateTime stored as ISO string
  assignedToUsers?: string[];   // Maps to AssignedToUsers in C# model
}

export enum ErrorLogSeverity {
  Info = 0,
  Notice = 1,
  Warning = 2,
  Error = 3,
  Fatal = 4
}

export enum ErrorLogState {
  New = 0,
  InProgress = 1,
  Resolved = 2,
  Closed = 3
}

export const errorSeverityOptions = [
  { value: 1, label: 'Info' },
  { value: 2, label: 'Notice' },
  { value: 3, label: 'Warning' },
  { value: 4, label: 'Error' },
  { value: 5, label: 'Fatal' }
];

export const errorStateOptions = [
  { value: 0, label: 'New' },
  { value: 1, label: 'In Progress' },
  { value: 2, label: 'Resolved' },
  { value: 3, label: 'Closed' }
];

File: EFDbContext/ErrorLog.cs
using System;
using System.Collections.Generic;

namespace AdventureWorksLT2019.EFDbContext;

/// <summary>
/// Audit table tracking errors in the the AdventureWorks database that are caught by the CATCH block of a TRY...CATCH construct. Data is inserted by stored procedure dbo.uspLogError when it is executed from inside the CATCH block of a TRY...CATCH construct.
/// </summary>
public partial class ErrorLog
{
    /// <summary>
    /// Primary key for ErrorLog records.
    /// </summary>
    public int ErrorLogId { get; set; }

    /// <summary>
    /// The date and time at which the error occurred.
    /// </summary>
    public DateTime ErrorTime { get; set; }

    /// <summary>
    /// The user who executed the batch in which the error occurred.
    /// </summary>
    public string UserName { get; set; } = null!;

    /// <summary>
    /// The error number of the error that occurred.
    /// </summary>
    public int ErrorNumber { get; set; }

    /// <summary>
    /// The severity of the error that occurred.
    /// </summary>
    public int? ErrorSeverity { get; set; }

    /// <summary>
    /// The state number of the error that occurred.
    /// </summary>
    public int? ErrorState { get; set; }

    /// <summary>
    /// The name of the stored procedure or trigger where the error occurred.
    /// </summary>
    public string? ErrorProcedure { get; set; }

    /// <summary>
    /// The line number at which the error occurred.
    /// </summary>
    public int? ErrorLine { get; set; }

    /// <summary>
    /// The message text of the error that occurred.
    /// </summary>
    public string ErrorMessage { get; set; } = null!;


    // Add these two properties:
    public string? Note { get; set; } // varchar(2000)
    public DateTime? LastUpdated { get; set; } // DateTime, nullable if you want

    public string? AssignedTo { get; set; } // Foreign key to AspNetUsers table

    public virtual AspNetUser? AssignedToUser { get; set; }
}
